pkg/api/methodHandlers.go

```diff
@@ -119,25 +121,39 @@ func (h *methodHandlers) execute(ctx context.Context, tracker attribute.Tracker,
 
 	// get a new context with the attribute bag attached
 	ctx = attribute.NewContext(ctx, ab)
-	var out *aspect.Output
+
+	numAdapters := len(h.configs[method])
+
+	// Loop over our configs, executing the adapter for each on another go routine. Because enqueue blocks until a worker
+	// is available it could block for a long time, so we check to see if our context was canceled each time we attempt
+	// to enqueue work.
+	results, enqueue := h.pool.requestGroup(h.mngr, ab, h.eval, numAdapters)
```

Comment:
The new handler takes an `Executor` interface and delegates execution to it.  (it is implemented by adapter manager). What if we change it to a BatchExecutor interface. which take []config.Combined

ExecuteBatch(ctx, []config.Combined, ...)

behind that batch can be a pool for parallel execution, or a non parallel model.
It would mean that pool code will live outside api.   pkg/pool, It will be constructed by  server.go and simply used by api.Handler